the-collab-lab/tcl-70-smart-shopping-list
# #15 Refactor user flow for Home Page, have manage and share icon following each list and provide instruction for user to use these new features. Meanwhile, follow our team wireframe design.

## Description



Refactor user flow, have manage and share icon following each list name and provide instruction for user to use these icons. Meanwhile, follow our team wireframe design.
<img width="500" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/d194e443-5c82-4599-a864-f5e45a7324cd">
For full user flow and wireframe design, see https://miro.com/app/board/uXjVKeyJp8o=/
## Related Issue

close #35 

## Acceptance Criteria

- [x] There should instructions for the user on how to view the list items (hint hint: tell them to click the list item name link)
- [ ] Change the list names from button to link (anchor tag)      
> _Used useNavigate instead of anchor tag_
- [x] Add a button next to list name link that takes the user to the manage list page
`_Have` a edit icon below name redirecting to manage list page_

Nice to have:
- [x]  If the list belongs to the user, there should be an option (like a button) to share the list. The button should not be there if the list was not created by the user (i.e. it was shared with the user)
> _Share icon added, temporary taking users to manage list page_

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: refactor     |

## Updates

### Before


<img width="500" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/bf4e2a4d-d4e7-4850-8a08-9dda27a10d57">

### After

<img width="500" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/b6509958-b784-41df-b8ed-fe44c722a452">

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull origin ag-refactor-home` and check that branch out with `git checkout ag-refactor-home` 
* Fill out the form with a new list. Make sure you receive successful message and a new list cardboard is added.
* Make sure the edit icon and share icon is clickable, it will direct you to manage list page.

## TODOs

- Add pop up form for share icon for user to enter a valid gmail
- Clear successful message after a certain time
- Add trash icon and delete list feature
- Decide Icon packs are design of the whole project, current icon used are placeholder
- Add credit to icon designer in an about page for our APP
- Design dark mode, current card background doesn't look good in dark mode
- I ellipsised successful message, however when listname is really long, user can't really tell whether the message is a success or fail
